jetty-ee10/jetty-ee10-webapp/src/main/java/org/eclipse/jetty/ee10/webapp/WebAppContext.java

```diff
@@ -1383,24 +1383,62 @@ public MetaData getMetaData()
         return _metadata;
     }
 
-    public static void addServerClasses(Server server, String... pattern)
+    /**
+     * @param server ignored.
+     * @param patterns the patterns to add
+     * @deprecated use {@link #addServerClasses(String...)} instead, will be removed in Jetty 12.1.0
```

Comment:
The method that doesn't take a server does a different job.    That sets the default for all servers rather than this method, which sets the default for a specific server.

What is the replacement for setting the default for a specific server?  That should be linked here.

Reply:
That's at the heart of what caused the issue.

The `Attributes` reference passed to `addServerClasses` was not used by the `WebAppContext.setServer(Server server)` call that needs the configuration.

Now with this PR the `add(Server|System)Classes` and `setServer(Server)` both use the environment specific `Attributes` locations.
Which works great, and even allows for wacky setups where 1 environment has a different setup than a different environment.

Do we have something like a jetty-core level `WebAppContext` (or something) that can be used as the `Server` level one?
I'm on the fence with the idea of having a `Server.add(Server|System)Classes` method which only has a `List<String>` (not the `ClassMatcher` option), but it feels like the wrong place for it.
Interestingly, while working this, I found a bug in the ee8 layer's XML where it attempted to use the `Server` instance to call `WebAppContext.addServerClasses(Server, String...)`, but due to XML order that `Server` didn't exist yet, which was simply worked around by using the Environment `Attributes` instead.

How about in a followup PR I move the 2 `ClassMatcher` impls (from ee10 & ee9) to jetty-util, and then put the `Attributes` neutral version in the new jetty-util code?

Reply:
@joakime note that the server classes are set on a static constant `ServletContextHandler.ENVIRONMENT`, so if you have 2 `Server`s in the same JVM, and you want different serverClasses configuration, you cannot.

Reply:
The ServletContextHandler.ENVIRONMENT (that exists in ee10, ee9, ee8) are JVM statics ATM (in HEAD).
I think changing that is out of scope for this PR, but I can see that possibly being a useful concept in the long run.